tests/test_playwright.py

```diff
@@ -42,7 +47,8 @@ def _build_test_site(site_name: str, sphinx_build_factory: Callable) -> None:
 
 def _check_test_site(site_name: str, site_path: Path, test_func: Callable):
     """Make the built test site available to Playwright, then run `test_func` on it."""
-    test_sites_dir.mkdir(exist_ok=True)
+    # Need to ensure parent directories exist in CI
+    test_sites_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
I ran into this as well while debugging; strictly speaking `parent=True` should not be necessary if the docs have been built. So I think in the event that the parent directories of `test_sites_dir` _don't_ exist, you've got a build problem that will make the tests fail anyway.

Reply:
> strictly speaking `parent=True` should not be necessary if the docs have been built [...] in the event that the parent directories of test_sites_dir don't exist, you've got a build problem that will make the tests fail anyway

Is the reason for keeping it that `site_path` isn't a child of `test_sites_dir`?  That would explain the symlinking a few lines below. (I know I wrote that code but I don't recall exactly why I did it that way)

Reply:
I thought there might be a reason for the symlink 🤷🏽‍♀️  but I do not see a reason why we would _have to_ keep this as is. I can go and refactor this.